### PR TITLE
Unify LEGATE_TEST and CUNUMERIC_TEST

### PR DESCRIPTION
**Pull request by bryevdv**
This PR removes usages of  `CUNUMERIC_TEST` in favor of single `LEGATE_TEST`. There is a corresponding `legate.core` PR to stop setting `CUNUMERIC_TEST` in the tester codepaths: https://github.com/nv-legate/legate.core/pull/715

### CODE REVIEW

File: cunumeric/runtime.py [hunk 1]
```diff
@@ -483,7 +484,7 @@ def is_eager_shape(self, shape: NdShape) -> bool:
         if volume == 0:
             return True
         # If we're testing then the answer is always no
-        if settings.test():
+        if legate_settings.test():
```
Comment (manopapad) [inline cunumeric/runtime.py:487]: Can we introduce a setting to control this, e.g. `disable_fallback`, and have it be controlled by `LEGATE_TEST` (false by default, true in testing mode)? Just to make sure this behavior is documented somewhere.

Comment (bryevdv) [reply cunumeric/runtime.py:487]: Can we try to decouple these (nothing depends on anything else) and switch to an affirmative boolean sense for enabling the behavior we want (simpler to reason about), e.g.
```
if settings.force_eager():  # CUNUMERIC_FORCE_EAGER
    return True
if legate_settings.test():
    return False
```
i.e. turn this into a question of precedence, with `CUNUMERIC_FORCE_EAGER` taking precedence? At least for me, that is much easier to reason about.

Comment (manopapad) [reply cunumeric/runtime.py:487]: That's fine, but it still leaves undocumented the fact that `LEGATE_TEST=1` forces arrays to be deferred. We could do it like this:

```
if settings.force_eager():  # CUNUMERIC_FORCE_EAGER
    return True
if settings.force_deferred():  # CUNUMERIC_FORCE_DEFERRED, has test_default=True
    return False
```

Or at that point maybe roll them all into a 3-state variable:

```
CUNUMERIC_FORCE_THUNK=eager/deferred/none
default = none
test_default = deferred
```

and change the code as follows:

```
    def is_eager_shape(self, shape: NdShape) -> bool:
        volume = calculate_volume(shape)
        # Special cases that must always be eager
        # Newly created empty arrays
        if volume == 0:
            return True
        # Arrays with more dimensions than what Legion was compiled for
        if len(shape) > LEGATE_MAX_DIM:
            return True
        if settings.force_thunk() == "eager":
            return True
        if settings.force_thunk() == "deferred":
            return False
        # no forcing; auto mode
        if len(shape) == 0:
            return self.max_eager_volume > 0
        # See if the volume is large enough
        return volume <= self.max_eager_volume
```

Comment (bryevdv) [reply cunumeric/runtime.py]: @manopapad check out https://github.com/nv-legate/cunumeric/pull/922/commits/46c9a38185563eba7ebbf35af69e11bf407bd9b5

Comment (manopapad) [reply cunumeric/runtime.py]: I think this works. Let's use this copy:

Force cuNumeric to always use a specific strategy for backing ndarrays: "deferred", i.e. managed by the Legate runtime, which enables distribution and accelerated operations, but has some up-front offloading overhead, or "eager", i.e. falling back to using a vanilla NumPy array. By default cuNumeric will decide this on a per-array basis, based on the size of the array and the accelerator in use.

Comment (bryevdv) [reply cunumeric/runtime.py]: https://github.com/nv-legate/cunumeric/pull/922/commits/9c79531f277f3ef944203a0c0bd8aaaccdc5651d

**Pull request merged**